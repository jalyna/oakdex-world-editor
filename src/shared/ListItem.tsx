import * as React from 'react'
import styled from 'styled-components'

import { DEFAULT_FONT, GREY_90, TEAL_70, GREY_30, TEAL_30 } from 'shared/theme'

interface ListItemProps {
  children: React.ReactNode,
  selected?: boolean
}

interface ItemProps {
  selected: boolean
}

const Item = styled.div`
  border-bottom: 1px solid ${GREY_90};
  padding: 5px;
  display: flex;
  width: 100%;
  box-sizing: border-box;
  align-items: center;
  background: ${({ selected }: ItemProps) => selected ? TEAL_70 : 'transparent'};
`

export const ItemTitle = styled.button`
  flex-grow: 2;
  flex-basis: 200%;
  outline: none;
  border: 0;
  background: transparent;
  padding: 4px;
  box-sizing: border-box;
  text-align: left;
  color: ${GREY_30}
  cursor: pointer;
  font-size: ${DEFAULT_FONT};
  font-size: 18px;

  &:hover {
    color: ${TEAL_30};
  }
`

export const Actions = styled.div `
  display: flex;
  flex-grow: 0;
`

export default function ListItem ({ children, selected }: ListItemProps) {
  return <Item selected={selected}>{children}</Item>
}
