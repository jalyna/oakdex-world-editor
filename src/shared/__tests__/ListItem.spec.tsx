import * as React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import ListItem, { Actions, ItemTitle } from '../ListItem'

describe('<ListItem>', () => {
  it('renders correctly', () => {
    const wrapper = mount(
      <ListItem>
        <ItemTitle>Foobar</ItemTitle>
        <Actions>Actions</Actions>
      </ListItem>
    )
    expect(toJson(wrapper, { noKey: false, mode: 'deep' })).toMatchSnapshot()
  })
})
