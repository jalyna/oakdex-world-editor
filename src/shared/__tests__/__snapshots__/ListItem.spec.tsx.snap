// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem> renders correctly 1`] = `
<styled.div>
  <div
    className="sc-bdVaJa cXMlYN"
  >
    <styled.button>
      <button
        className="sc-bwzfXH ggNpSn"
      >
        Foobar
      </button>
    </styled.button>
    <styled.div>
      <div
        className="sc-htpNat dGbXB"
      >
        Actions
      </div>
    </styled.div>
  </div>
</styled.div>
`;
